redis/tests/support/mock_cluster.rs

```diff
@@ -27,10 +30,41 @@ use tokio::runtime::Runtime;
 
 type Handler = Arc<dyn Fn(&[u8], u16) -> Result<(), RedisResult<Value>> + Send + Sync>;
 
-static HANDLERS: Lazy<RwLock<HashMap<String, Handler>>> = Lazy::new(Default::default);
+#[derive(Default)]
+pub struct MockConnectionUtils {
+    pub handler_map: HashMap<String, Handler>,
+    pub connection_id_provider: AtomicUsize,
```

Comment:
the new fields should be per-handler, not global. hopefully one day we'll manage to make the tests run concurrently, and then this will be an issue.